libraries/testbot/tests/mainDialog.tests.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+const { DialogTestClient, DialogTestLogger } = require('botbuilder-testing');
+const { MainDialog } = require('../dialogs/mainDialog');
+const assert = require('assert');
+
+
+class MockLuisHelper {
+
+    static async executeLuisQuery(logger, context) {
+
+        return {
+            origin: 'Paris',
+            destination: 'Berlin',
+            travelDate: '2022-02-22',
+        }
+    }
+}
+
+describe('mainDialog', function() {
+   
+    it('should create a DialogTestClient', async function() {
+        let client = new DialogTestClient();
+        assert(client instanceof DialogTestClient, 'Created an invalid object');
+    });
+
+    let tests = require('./testData/mainDialogTestCases.js');
+
+    let dialog = new MainDialog(null, MockLuisHelper);
```

Comment:
Pass in bookingDialog mock to avoid calling child dialog in tests.

MainDialog test cases can be found here: https://github.com/microsoft/BotBuilder-Samples/blob/8dcaafc51a73f6663c7f53d07f3c9eaec0b68d5d/samples/csharp_dotnetcore/13.core-bot.tests/Dialogs/MainDialogTests.cs

Reply:
Done